runtime/queries/metricsview_comparison_toplist.go

```diff
@@ -113,12 +115,69 @@ func (q *MetricsViewComparison) Resolve(ctx context.Context, rt *runtime.Runtime
 	}
 
 	if !isTimeRangeNil(q.ComparisonTimeRange) {
+		// execute toplist for base and get dim list
+		// create and add filter and execute comprison toplist
+		// remove strict limits in comp toplist sql
+		if drivers.DialectDruid != olap.Dialect() || q.NoExploreExactifyMode {
+			return q.executeComparisonToplist(ctx, olap, mv, priority, security)
+		}
+
+		if q.Exact {
+			return fmt.Errorf("Cannot calculate Exact when Druid Exactify is requested")
+		}
+
+		// Druid-based `exactify` approach:
+		// The previous query collected a list of dimensions. The following query will filter in those dimensions
+		// to reduce the topN list to fit in memory to exclude discrepancies due incomplete topN tables merged by the previous query
+		if q.isBase() || q.isDeltaComparison() {
+			err = q.executeToplist(ctx, olap, mv, priority, security)
```

Comment:
list of measures can be optimized here, only compute the measures which are used in sort order. Only do it if we have more than 4 measures.

Reply:
this is a good query for all leaderboards, for expanded table it will depend on the number of measures.

Reply:
done